app/models/canned_response.rb

```diff
@@ -14,4 +14,15 @@ class CannedResponse < ApplicationRecord
 
   validates_presence_of :name_en
   validates_presence_of :content_en
+  validates :guid, uniqueness: true, allow_nil: true
+
+  before_destroy :validate_for_private_messages
+
+  scope :by_private, ->(is_private) { where(is_private: is_private) }
+
+  private
+
+  def validate_for_private_messages
```

Comment:
Can you explain what this is for?

Reply:
A validation to ensure no deletion happens for private message

`before_destroy: validate_for_private_messages`